subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/report/HtmlDependencyVerificationReportRenderer.java

```diff
@@ -0,0 +1,446 @@
+        registerModal("checksum-mismatch", "Incorrect checksum",
+            "        <p>This indicates that the dependency verification file <b>failed</b> because the actual checksum of the dependency artifact didn't match the expected checksum declared in the verification metadata.</p>",
+            "        <p>This happens when a dependency was compromised or that downloaded artifact isn't the one that you expected.</p>",
+            "        <p>It's important that you <b>carefully review this problem</b>.</p>");
+        registerModal("deleted-artifact", "Deleted artifact",
+            "        <p>This error usually indicated that the local dependency cache was tampered with.</p>",
+            "        <p>This happens when someone manually deletes an artifact from the Gradle dependency cache, which is now corrupt.</p>");
+        registerModal("signature-file-missing", "Missing signature file",
+            "        <p>The signature file for this artifact wasn't found.</p>",
+            "        <p>Usually it indicates that the signature doesn't exist in the repository the artifact was downloaded from.</p>",
+            "        <p>In general this is not a problem but you should then declare at least one checksum for verification to pass.</p>");
+    }
+
+    private void copyReportResource(File outputDirectory, String dirName, String fileName) {
+        File targetDir = new File(outputDirectory, dirName);
+        targetDir.mkdirs();
+        copyResource(fileName, new File(targetDir, fileName));
+    }
+
+    private void copyResource(String name, File outputFile) {
+        try (InputStream in = getClass().getResourceAsStream(name)) {
+            Files.copy(in, outputFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void prerenderSection(Section section) {
+        int size = section.errors.size();
+        contents.append("<a class=\"uk-accordion-title\" href=\"#\"><span uk-icon=\"chevron-right\"></span>").append(section.title)
+            .append("&nbsp;<span class=\"uk-badge\">")
+            .append(size)
+            .append(size < 2 ? " error" : " errors")
+            .append("</span></a>");
+    }
+
+    private void renderSection(Section section) {
+        contents.append("<div class=\"uk-accordion-content\">")
+            .append("<table class=\"uk-table uk-table-hover uk-table-divider uk-table-middle uk-table-small\">\n")
+            .append("    <thead>\n")
+            .append("        <tr>\n")
+            .append("            <th class=\"uk-table-shrink uk-width-auto uk-text-nowrap\">Module</th>\n")
+            .append("            <th class=\"uk-table-shrink uk-width-auto uk-text-nowrap\">Artifact</th>\n")
+            .append("            <th class=\"uk-table-shrink uk-width-expand\">Problem</th>\n")
```

Comment:
There can be multiple problems displayed per cell
```suggestion
            .append("            <th class=\"uk-table-shrink uk-width-expand\">Problem(s)</th>\n")
```